Fix up missing 'Wix4' prefixes on MSMQ Permission tables.

bevanweiss opened the pull request:
Fix up missing 'Wix4' prefixes on MSMQ Permission tables.
And updated decompiler to Wix4 table names etc in line with firewall CA.

Fixes wixtoolset/issues#8902

robmen commented:
I rebased on `main` with all the other PRs in and this PR now fails with:

```
WixRunner failed with exit code: 1059
Output: Temp\.WIXTEST ea4bxgeg.d3m\bin\test.msi : error TEST1059: The Wix6Group tabl
e contains a row with primary key(s) 'TestGroup' whose Group_ column contains a value, 'TestGroup', which specifie
s a foreign key relationship with the Group table. However, since the expected foreign row specified by this value
does not exist, this will result in some information being left out of the decompiled output.
```

I'd hoped rebasing would straighten things out, but it looks like it only made it worse. 😄

bevanweiss commented:
> I rebased on `main` with all the other PRs in and this PR now fails with:
> 
> ```
> WixRunner failed with exit code: 1059
> Output: Temp\.WIXTEST ea4bxgeg.d3m\bin\test.msi : error TEST1059: The Wix6Group tabl
> e contains a row with primary key(s) 'TestGroup' whose Group_ column contains a value, 'TestGroup', which specifie
> s a foreign key relationship with the Group table. However, since the expected foreign row specified by this value
> does not exist, this will result in some information being left out of the decompiled output.
> ```
> 
> I'd hoped rebasing would straighten things out, but it looks like it only made it worse. 😄

I've just rebased also, and fixed up the merge conflict around the Group decompiler (which might have been the trigger for the error you saw).
I'm in the slow process of building it now (locally).  So should have it fixed up 'shortly'.

bevanweiss commented:
'Solved'..
It was a Wix6Group decompiler issue.  That was only now being stretched with the msmq round trip decompiler test.
This is hopefully good to go now.

Adding some more decompiler round trip tests seems worthwhile for the various extensions.  So I'll look to run a few in on the weekend.
Those unit tests do add time to the overall build time though.
Perhaps there should be some flags to do pure compile builds, and then each area (Core vs individual extension) could be enabled  (for testing) separately.  With one overall 'full test' that sets all the flags.

robmen commented:
In general, we've had a good bit of debate about how much value the extension decompilers provide. Decompiling is a great way to move from a non-WiX-based installation technology to WiX. However, decompiling WiX-based MSIs is not a scenario because you already have .wxs code. The fact that most of the extension decompilers were not ported from v3 and we've had no complaints about them suggests that our choice to invest in the converter for upgrading .wxs code was the right call and that extension decompilers may be unnecessary.

Decompiling has a maintenance cost, and before spending more time in that direction, we should decide if it's really worth it.

bevanweiss commented:
```
Creating library D:\a\wix\wix\build\Util.wixext\Debug\x86\utilca.lib and object D:\a\wix\wix\build\Util.wixext\Debug\x86\utilca.exp
  utilca.vcxproj -> D:\a\wix\wix\build\Util.wixext\Debug\x86\utilca.dll
D:\a\wix\wix\src\ext\Util\wixlib\UtilBootstrapperExtension_Platform.wxi(8): error WIX0103: Cannot find the WixBundlePayload file '!(bindpath.utilbe.arm64)utilbe.dll'. The following paths were checked: !(bindpath.utilbe.arm64)utilbe.dll, D:\a\wix\wix\build\Util.wixext\Debug\arm64\utilbe.dll [D:\a\wix\wix\src\ext\Util\wixlib\util.wixproj]
D:\a\wix\wix\src\ext\Util\wixlib\UtilBootstrapperExtension_Platform.wxi(8): error WIX0103: Cannot find the WixBundlePayload file '!(bindpath.utilbe.x64)utilbe.dll'. The following paths were checked: !(bindpath.utilbe.x64)utilbe.dll, D:\a\wix\wix\build\Util.wixext\Debug\x64\utilbe.dll [D:\a\wix\wix\src\ext\Util\wixlib\util.wixproj]
D:\a\wix\wix\src\ext\Util\wixlib\UtilExtension_Platform.wxi(399): error WIX0103: Cannot find the Binary file '!(bindpath.utilca.arm64)utilca.dll'. The following paths were checked: !(bindpath.utilca.arm64)utilca.dll, D:\a\wix\wix\build\Util.wixext\Debug\arm64\utilca.dll [D:\a\wix\wix\src\ext\Util\wixlib\util.wixproj]
D:\a\wix\wix\src\ext\Util\wixlib\UtilExtension_Platform.wxi(399): error WIX0103: Cannot find the Binary file '!(bindpath.utilca.x64)utilca.dll'. The following paths were checked: !(bindpath.utilca.x64)utilca.dll, D:\a\wix\wix\build\Util.wixext\Debug\x64\utilca.dll [D:\a\wix\wix\src\ext\Util\wixlib\util.wixproj]
```

Hmm.. back to this error.  Which I'm not seeing locally.
It only appears to be complaining about the x64 and arm64 items (utilbe.dll and utilca.dll).
So x86 dlls got built correctly perhaps...


I'm thinking it's caused by this Util reference in the Msmq UnitTests (NOT integration test).
I'm probably doing this reference wrong (i.e. the unit test shouldn't reference external dependencies in this manner).
```xml
  <ItemGroup>
    <ProjectReference Include="..\..\wixext\WixToolset.Msmq.wixext.csproj" />
    <ProjectReference Include="..\..\..\Util\wixext\WixToolset.Util.wixext.csproj" />
  </ItemGroup>
```

bevanweiss commented:
> EDIT: I've only been building in 'debug' using `.\devbuild.cmd` without parameters. Buildbot does a release build. That does reveal the difference, so now I get the same failure locally. Why debug works, and release doesn't... unsure. I'm still very confident it's about the ProjectReference and this resulting in only the x86 util assemblies being generated.

I haven't really made progress :(  I seem to keep circling this same dependency build issue.
It's definitely about needing both msmq and util in this msmq test.  For some reason the util appears to be getting built in debug mode (instead of release).  Any ideas?

barnson commented:
I've been working on converting ext unit tests to MSTest and NetFx references Util via `<PackageReference Include="WixToolset.Util.wixext" />`.

bevanweiss commented:
> I've been working on converting ext unit tests to MSTest and NetFx references Util via `<PackageReference Include="WixToolset.Util.wixext" />`.

I'll give it a go replicating how NetFx currently does it.

I am surprised that it 'just worked' under the top level Debug build though.
Only on the top level Release build does it fail, but where the lower level Util build was occuring in Debug config... which i would have expected the top level Release config to be pushed into the lower Util build (and it shouldn't have even needed to build Util again)..

bevanweiss commented:
I've disabled the decompile test, so it should be good to go for the 'expected' functionality.

I'll likely need to review the decompile functionality of the Wix4Group / Wix6Group stuff first, to get that all working correctly.
Which might then just sort out this msmq decompile issue.
